R/deltaccd_plot.R

```diff
@@ -143,13 +155,16 @@ plotHeatmap = function(geneNames, emat, groupVec = NULL) {
 plotRefHeatmap = function(refCor) {
   if (any(rownames(refCor) != colnames(refCor)) || !isSymmetric(refCor)) {
     stop('refCor must be a correlation matrix, with identical rownames and colnames.')}
-
-  df = data.frame(refCor, gene1 = geneNames, stringsAsFactors = FALSE,
-                  check.names = FALSE) %>%
-    tidyr::gather(-gene1, key = gene2, value = rho) %>%
-    dplyr::filter(gene1 != gene2) %>%
-    dplyr::mutate(gene1 = factor(gene1, rownames(ematNow)),
-                  gene2 = factor(gene2, rev(rownames(ematNow))))
-
+  
+  dt = data.table::data.table(refCor, gene1 = geneNames)
+  
+  dt = data.table::melt
+  dt = data.table::melt(dt1, id.vars = 'gene1', measure.vars = 'gene2'
+    , value.name = 'rho')
+  dt = dt[gene1 != gene2]
+  dt[
+    , `:=`(gene1 = factor(gene1, rownames(ematNow))
+           , gene2 = factor(gene2, rev(rownames(ematNow))))]
+ 
   cLims = calcColorLimits(df$rho)
```

Comment:
These lines are still using df instead of dt. Good example of not having a test.